src/main/java/org/generationcp/middleware/pojos/GermplasmStudySource.java

```diff
@@ -0,0 +1,119 @@
+package org.generationcp.middleware.pojos;
+
+import org.generationcp.middleware.pojos.dms.DmsProject;
+import org.generationcp.middleware.pojos.dms.ExperimentModel;
+import org.generationcp.middleware.service.api.study.germplasm.source.GermplasmStudySourceInput;
+import org.pojomatic.Pojomatic;
+import org.pojomatic.annotations.AutoProperty;
+
+import javax.persistence.Basic;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "germplasm_study_source")
+@AutoProperty
+public class GermplasmStudySource {
+
+	public GermplasmStudySource(final Germplasm germplasm, final DmsProject study, final ExperimentModel experimentModel,
+		final GermplasmStudySourceType germplasmStudySourceType) {
+		this.germplasm = germplasm;
+		this.study = study;
+		this.experimentModel = experimentModel;
+		this.germplasmStudySourceType = germplasmStudySourceType;
+	}
+
+	public GermplasmStudySource(final GermplasmStudySourceInput input) {
+		this.germplasm = new Germplasm(input.getGid());
+		this.study = new DmsProject(input.getStudyId());
+		this.experimentModel = new ExperimentModel(input.getObservationUnitId());
+		this.germplasmStudySourceType = input.getType();
+	}
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	@Basic(optional = false)
+	@Column(name = "source_id")
+	private Integer sourceId;
```

Comment:
We did not have a standard for naming ids in the past, but we are tending to use <table_name> + _id

Reply:
Done